src/main/java/edu/cornell/kfs/sys/service/impl/ISOFIPSConversionServiceImpl.java

```diff
@@ -0,0 +1,74 @@
+package edu.cornell.kfs.sys.service.impl;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Iterator;
+import java.util.List;
+
+import org.apache.log4j.LogManager;
+import org.apache.log4j.Logger;
+
+import edu.cornell.kfs.sys.businessobject.ISOFIPSCountryMap;
+import edu.cornell.kfs.sys.dataaccess.ISOFIPSCountryMapDao;
+import edu.cornell.kfs.sys.service.ISOFIPSConversionService;
+
+public class ISOFIPSConversionServiceImpl implements ISOFIPSConversionService {
+    
+    private static final Logger LOG = LogManager.getLogger(ISOFIPSConversionServiceImpl.class);
+    
+    private ISOFIPSCountryMapDao isoFipsCountryMapDao;
+    
+    /*
+     * ISO-to-FIPS mappings are many-ISO-to-one-FIPS.
+     */
+    public String convertISOCountryCodeToFIPSCountryCode(String isoCountryCode) {
+        List <ISOFIPSCountryMap> mappingsFound = new ArrayList<ISOFIPSCountryMap>();
+        Collection <ISOFIPSCountryMap> fipsCodesFound = isoFipsCountryMapDao.findFipsCountries(isoCountryCode);
+        
+        for (Iterator iter = fipsCodesFound.iterator(); iter.hasNext();) {
+            ISOFIPSCountryMap fipsCodeMapping = (ISOFIPSCountryMap) iter.next();
+            mappingsFound.add(fipsCodeMapping);
+        }
+        
+        if (mappingsFound.isEmpty()) {
+            LOG.error("convertISOCountryCodeToFIPSCountryCode: No ISO-to-FIPS Country mapping found for ISO Country code : " + isoCountryCode);
+            throw new RuntimeException("No ISO-to-FIPS Country mapping found for ISO Country code : " + isoCountryCode);
+        } else if (mappingsFound.size() > 1) {
```

Comment:
Thanks for the explanation. I'm fine with further functional or technical adjustments being handled in future user stories.

Reply:
I thought about this a little more and I do have a concern that future user stories will put the logic that should be in their service implementation class (i.e. Jaggaer, Tax, PaymentWorks,...) into this generic data lookup class making this generic solution too specific thus tying it directly to those other implementations which we do not want.

Suggestion : What if I did the following? 

1. Created 4 exceptions that were extended from RuntimeException (NoISOtoFIPSMappingException, ManyISOtoFIPSMappingException, NoFIPStoISOMappingException, ManyFIPStoISOMappingException).

2. Created two new methods that returned the list of country code mappings when many were found (findManyISOCountryCodesForFIPSCode, findManyFIPSCountryCodesForISOCode).

This would allow the calling service implementations (i.e. Jaggaer, Tax, PaymentWorks,...) to identify those specific conditions when they happened, have the ability to request the data they need, and have specific/unique logic to deal with those anomalies for that part of the functionality when those exceptions occurred since the logic may be different for each area.   Thoughts??

Reply:
Yes, the solution you suggested sounds like a good idea. Thanks!